### PR TITLE
Add migration for rendered_map_index column in task_instance table.

### PR DESCRIPTION
**Pull request by tirkarthi**
Add migration for rendered_map_index column added to task_instance table in PR #36797

related: #36797

### CODE REVIEW

File: airflow/migrations/versions/0134_2_9_0_add_rendered_map_index_to_taskinstance.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+def upgrade():
+    """Apply Add rendered_map_index to TaskInstance."""
+    conn = op.get_bind()
+
+    with op.batch_alter_table("task_instance") as batch_op:
+        if conn.dialect.name == "mysql":
+            batch_op.add_column(sa.Column("rendered_map_index", sa.Text(length=64), nullable=True))
+        else:
+            batch_op.add_column(sa.Column("rendered_map_index", sa.String(length=64), nullable=True))
```
Comment (uranusjr) [inline airflow/migrations/versions/0134_2_9_0_add_rendered_map_index_to_taskinstance.py]: Why do we need this? Either using Text or String for all databases is reasonable to me.

Comment (tirkarthi) [reply airflow/migrations/versions/0134_2_9_0_add_rendered_map_index_to_taskinstance.py]: I just copy pasted from my previous migration PR. I have seen this pattern elsewhere in the migration files but they are mostly for columns storing longer strings. I guess for this case of 64 characters String is enough. Thanks.

https://github.com/apache/airflow/blob/main/airflow/migrations/versions/0117_2_4_0_add_processor_subdir_to_dagmodel_and_.py

https://github.com/apache/airflow/blob/f21e641193ce559b42ddc64585ee4e1a9c15799c/airflow/migrations/versions/0121_2_5_0_add_dagrunnote_and_taskinstancenote.py#L68-L70

**Review by jedcunningham — APPROVED** [review]

**Comment by jedcunningham** [comment]
Didn't we have a test to ensure migrations and models always build the same db? Wonder when that broke...

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Just a watch-out - we have two migrations in-progress. I think we have a protection that will make the secon one not mergable if the first is but there will be some re-work needed once the other mighration is merged. https://github.com/apache/airflow/pull/37731

**Comment by uranusjr** [comment]
It’s interesting to me this does not show up in tests that use `rendered_map_index` though. SQLAlchemy is more permissive than I imagined…

**Pull request merged**